packages/augur-sdk/src/state/db/MarketDB.ts

```diff
@@ -186,4 +206,102 @@ export class MarketDB extends DerivedDB {
+      return this.processInitialReportSubmitted(log);
+    } else if (log.name === 'DisputeCrowdsourcerCompleted') {
+      return this.processDisputeCrowdsourcerCompleted(log);
+    } else if (log.name === 'MarketFinalized') {
+      return this.processMarketFinalized(log);
+    }
+    return log;
+  }
+
+  private processMarketCreated(log: ParsedLog): ParsedLog {
+    log['reportingState'] = MarketReportingState.PreReporting;
+    log['invalidFilter'] = false;
+    log['marketOI'] = '0x00';
+    log['volume'] = '0x00';
+    log['disputeRound'] = '0x00';
+    log['totalRepStakedInMarket'] = '0x00';
+    log['hasRecentlyDepletedLiquidity'] = false;
+    log['feeDivisor'] = new BigNumber(1).dividedBy(new BigNumber(log['feePerCashInAttoCash'], 16).dividedBy(QUINTILLION)).toNumber();
+    return log;
+  }
+
+  private processInitialReportSubmitted(log: ParsedLog): ParsedLog {
+    log['reportingState'] = MarketReportingState.CrowdsourcingDispute;
+    log['totalRepStakedInMarket'] = log['amountStaked']
+    log['tentativeWinningPayoutNumerators'] = log['payoutNumerators']
+    log['disputeRound'] = '0x1';
+    return log;
+  }
+
+  private processDisputeCrowdsourcerCompleted(log: ParsedLog): ParsedLog {
+    const pacingOn: boolean = log['pacingOn'];
+    log['reportingState'] = pacingOn ? MarketReportingState.AwaitingNextWindow : MarketReportingState.CrowdsourcingDispute;
+    log['tentativeWinningPayoutNumerators'] = log['payoutNumerators']
+    return log;
+  }
+
+  private processMarketFinalized(log: ParsedLog): ParsedLog {
+    log['reportingState'] = MarketReportingState.Finalized;
+    log['finalizationBlockNumber'] = log['blockNumber'];
+    log['finalizationTime'] = log['timestamp'];
+    return log;
+  }
+
+  processNewBlock = async (block: Block): Promise<void> => {
+    const timestamp = (await this.augur.getTimestamp()).toNumber();
+    await this.processTimestamp(timestamp, parseInt(block.number))
+  }
+
+  processTimestampSet = async (log: TimestampSetLog): Promise<void> => {
+    const timestamp = new BigNumber(log.newTimestamp).toNumber();
+    await this.processTimestamp(timestamp, log.blockNumber)
+  }
+
+  private async processTimestamp(timestamp: number, blockNumber: number): Promise<void> {
+    const eligibleMarketDocs = await this.find({
+      selector: {
+        reportingState: { $in: [
+          MarketReportingState.PreReporting,
+          MarketReportingState.DesignatedReporting,
+          MarketReportingState.CrowdsourcingDispute,
+          MarketReportingState.AwaitingNextWindow,
+        ] }
+      }
+    });
+    const eligibleMarketsData = eligibleMarketDocs.docs as unknown as MarketData[];
+    const updateDocs = [];
+
+    for (const marketData of eligibleMarketsData) {
+      let reportingState: MarketReportingState = null;
+      const marketEnd = new BigNumber(marketData.endTime, 16);
+      const openReportingStart = marketEnd.plus(SECONDS_IN_A_DAY);
```

Comment:
marketEnd.plus(SECONDS_IN_A_DAY); should this be plus one second to "officially" be in "open reporting"?

Reply:
technically true! Will do